src/libraries/System.Linq.Queryable/src/System/Linq/CachedReflection.cs

```diff
@@ -843,6 +843,13 @@ internal static class CachedReflectionInfo
              (s_Zip_TFirst_TSecond_TResult_3 = new Func<IQueryable<object>, IEnumerable<object>, Expression<Func<object, object, object>>, IQueryable<object>>(Queryable.Zip).GetMethodInfo().GetGenericMethodDefinition()))
               .MakeGenericMethod(TFirst, TSecond, TResult);
 
+        private static MethodInfo? s_Zip_TFirst_TSecond_TThird_3;
+
+        public static MethodInfo Zip_TFirst_TSecond_TThird_3(Type TFirst, Type TSecond, Type TThird) =>
+             (s_Zip_TFirst_TSecond_TThird_3 ??
+             (s_Zip_TFirst_TSecond_TThird_3 = new Func<IQueryable<object>, IEnumerable<object>, IEnumerable<object>, IQueryable<(object, object, object)>>(Queryable.Zip).GetMethodInfo().GetGenericMethodDefinition()))
+              .MakeGenericMethod(TFirst, TSecond, TThird);
```

Comment:
I was going to comment saying `??=` could be used to simplify this, but I see you're just following the pattern used in a bunch of other cases above.  We can clean it up later.